### ISSUE TITLE
Error generating schema using serverless, webpack and typegoose

### ISSUE THREAD

**Issue by visheshd** [19majkel94/type-graphql#367]
**Describe the bug**
A basic setup of type-graphql with typegoose

**To Reproduce**
https://github.com/visheshd/typegraphql-schema-error

**Expected behavior**
Error on schema generation

**Logs**
```
(node:67822) UnhandledPromiseRejectionWarning: Error: Generating schema error
    at Function.<anonymous> (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:151897:27)
    at Generator.next (<anonymous>)
    at fulfilled (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:149469:58)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

```
**Enviorment (please complete the following information):**
 - OS: [e.g. Windows]
 - Node (e.g. 8.10.0)
 - Package version [e.g. 0.17.4] (please check if the bug still exist in newest release)
 - TypeScript version 3.4.5
- Typegoose version
- serverless
- serverless-webpack

**Comment by 19majkel94** [comment]
Please catch the error and print `originalError` property.

**Comment by visheshd** [comment]
Added more logs. @19majkel94 thanks for looking into it!

Original error while calling buildSchema

```error: Error occured while boostrapping: , {
  "details": [
    {
      "message": "Type Query must define one or more fields."
    }
  ]
}```

**Comment by 19majkel94** [comment]
I think that calling `(global as any).TypeGraphQLMetadataStorage!.clear()` is not a good idea as it clears all resolver metadata, so no queries defined.

**Comment by visheshd** [comment]
Added it as a fix from one of the issue comments earlier. Removing it produces another error.

```Cannot determine GraphQL output type for _id
    at Function.getGraphQLOutputType (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:152290:19)
    at objectType.fields.reduce (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:152070:44)
    at Array.reduce (<anonymous>)
    at fields (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:152064:56)
    at resolveThunk (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:73038:40)
    at defineFieldMap (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:73218:18)
    at GraphQLObjectType.getFields (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:73175:27)
    at typeMapReducer (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:75099:106)
    at typeMapReducer (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:75109:22)
    at Array.reduce (<anonymous>)
    at new GraphQLSchema (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:74926:28)
    at Function.generateFromMetadataSync (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:151927:24)
    at Function.<anonymous> (/Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:151912:33)
    at Generator.next (<anonymous>)
    at /Volumes/data/Workspace/typegraphql-error/.webpack/service/handler.js:149491:71```

It looks like it there is an issue with ObjectIDScalar ... I basically copied that and the pattern of defining the id field from the `Typegoose` example.

**Comment by visheshd** [comment]
Turns out I was importing the wrong the ObjectId class in the handler vs everywhere else. Causing the scalar to fail. 

It works now but I had to remove the emit schema flag. With the flag it throws

```
(node:76034) UnhandledPromiseRejectionWarning: Error: EACCES: permission denied, open '/schema.graphql'
(node:76034) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:76034) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by visheshd** [comment]
This might acutally be a bug

```javascript
// emitSchemaFile: path.resolve(__dirname, 'schema.graphql'), for some reason specifying path fails because it tries to write it to '/schema.graphql'

emitSchemaFile: {}, // this forces it to use the default path and works.
```

**Comment by 19majkel94** [comment]
You might have no write access in serveless environment - try  `path.resolve(process.cwd(), 'schema.graphql')`.

**Comment by visheshd** [comment]
The plain resolvers returning a primitive types are working. But the server is timing out any database call using Typegoose. 

They are written like the example but nothing below the following line gets executed. It simply times out.
https://github.com/visheshd/typegraphql-schema-error/blob/master/src/user/UserResolver.ts#L34

**Comment by 19majkel94** [comment]
¯\_(ツ)_/¯

Please create a reproducible minimal code example that I can debug. Whole repository where I have to dig is not helpful. Try to isolate the code parts that have problem and reduce the complexity of setup.

**Comment by visheshd** [comment]
The repo was fairly minimal but further removed few files that were unnecessary. The repository has nothing but the code from the examples. The readme has minimal instruction on how to reproduce the issue.

```javascript
  @Query(_returns => [User], { nullable: true })
  async users(): Promise<User[]>{
         
    log.debug(`Fetching users from the database. Models: ${JSON.stringify(mongoose.connection.modelNames())}`);
    try {
      
      const users = await UserModel.find({}); // >>> execution stalls here
      log.debug(`Users from the database: ${JSON.stringify(users)}`)
      return users;
    } catch(e) {
      log.debug(`Database Error: ${JSON.stringify(e)}`);
      return [];
    } finally {
      log.debug(`Finishing up call to the database`);
    }
  }
```

I found one open issue on Typegoose loosely related to this but the work arounds suggested didn't work https://github.com/szokodiakos/typegoose/issues/64

**Comment by 19majkel94** [comment]
Try replace it with `Promise.resolve` - if it works, it's not TypeGraphQL related issue

**Comment by visheshd** [comment]
Sure will give it another go tomorrow. I replaced the code with just things mentioned in the examples and I am still able to reproduce the issue. Is the typegoose example working for you locally?

```javascript
@Query(() => [Recipe])
  async recipes(): Promise<Recipe[]> {
    return Promise.resolve(RecipeModel.find({}));  // >> stuck here with or without Promise.resolve
  }
```

**Comment by 19majkel94** [comment]
```ts
  @Query(() => [Recipe])
  async recipes(): Promise<Recipe[]> {
    return Promise.resolve([]);
  }
```

**Comment by visheshd** [comment]
But you've not used `RecipeModel`? 
I am sorry but I didn't get your point?
```javascript
 @Query(() => [Recipe])
  async recipes(): Promise<Recipe[]> {
    return await Promise.resolve(RecipeModel.find({})); 
  }
```
also times out.

**Comment by 19majkel94** [comment]
> I am sorry but I didn't get your point?

When we debug, we try to remove variables from our equation.

So by replacing `RecipeModel.find` with `Promise.resolve` we simulate async resolvers as the execution stuck with Typegoose and only "The plain resolvers returning a primitive types are working".

**Comment by visheshd** [comment]
I think that is what I did to rule out things with primitives but sure. The Promise.resolve([]) version works. It is definitely was a typegoose issue. Had to pass the existingConnection.

I had to 

```javascript 

export let RecipeModel: Model<InstanceType<Recipe>>;

(async () => {
  RecipeModel = new Recipe().getModelForClass(Recipe, {
    existingConnection: await getConnection()
  });
})();

```
For anyone who is bumping into this in the future here is the [link to the repo that is working](https://github.com/causecode/type-graphql-typegoose-serverless).
